src/main/features/response/form/models/timeline.ts

```diff
@@ -0,0 +1,68 @@
+import { ValidateNested } from 'class-validator'
+
+import { Serializable } from 'models/serializable'
+import { TimelineRow } from 'response/form/models/timelineRow'
+
+export const INIT_ROW_COUNT: number = 4
+
+export const MAX_NUMBER_OF_EVENTS: number = 20
+
+export class Timeline implements Serializable<Timeline> {
+  readonly type: string = 'breakdown'
+
+  @ValidateNested({ each: true })
+  rows: TimelineRow[]
+
+  constructor (rows: TimelineRow[] = Timeline.initialRows()) {
+    this.rows = rows
+  }
+
+  static fromObject (value?: any): Timeline {
+    if (!value) {
+      return value
+    }
+
+    return new Timeline(value.rows ? value.rows.map(TimelineRow.fromObject) : [])
+  }
+
+  private static initialRows (rows: number = INIT_ROW_COUNT): TimelineRow[] {
+    return new Array(rows).fill(TimelineRow.empty())
+  }
+
+  deserialize (input?: any): Timeline {
+    if (input) {
+      this.rows = this.deserializeRows(input.rows)
+    }
+
+    return this
+  }
+
+  appendRow () {
+    if (this.rows.length < MAX_NUMBER_OF_EVENTS) {
+      this.rows.push(TimelineRow.empty())
+    }
+  }
+
+  clearUselessRows () {
```

Comment:
I think "Useless" is not the most accurate word here (https://translate.google.com/#en/pl/useless), what do you think about using "Blank" instead?

Reply:
How about "[Superfluous](https://translate.google.com/#en/pl/superfluous)"?